src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -492,6 +492,355 @@ TEST(Aktualizr, FullWithUpdatesNeedReboot) {
   }
 }
 
+#ifdef FIU_ENABLE
+
+#include <fiu-control.h>
+#include "utilities/fault_injection.h"
+
+class HttpInstallationFailed : public HttpFake {
+ public:
+  HttpInstallationFailed(const boost::filesystem::path& test_dir_in) : HttpFake(test_dir_in, "hasupdates") {}
+
+  HttpResponse handle_event(const std::string& url, const Json::Value& data) override {
+    (void)url;
+
+    for (auto& event_data : data) {
+      auto event_id = event_data["eventType"]["id"].asString();
+
+      if (event_id == "EcuInstallationCompleted") {
+        install_completion_status_ = event_data["event"]["success"].asBool();
```

Comment:
I think there should be an `EXPECT_FALSE` here if possible. We need to explicitly check that the reports we send have the values we expect, not just that what we read out of the database is what we expect. The low level stuff is worth checking, but it's the high level that I'm actually most concerned with.

Reply:
Yes, it implicitly tested by changing the flavor value. If the value is not correct/expected then the multiple assertions/expectation will fail.
But, yes I agree on adding explicit check.